### ISSUE TITLE
MidSessionStatusPoller.pollForFinalSessionStatus hangs when internet connection drops.

### ISSUE THREAD

**Issue by ErkiHallasoo** [SK-EID/mid-rest-java-client#10]
**Do you want to request a feature or report a bug?**
Report a bug.

**What is the current behavior?**
When internet connection drops during `MidConnector.getSessionStatus`, no response or exception get returned and code execution hangs.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
*Step 1: Start authentication process.*
*Step 2: When your phone gets the notification, drop your internet connection.*
If you attach debugger on **line 71**  in `MidSessionStatusPoller.class`, you will notice that java is waiting for a response from `this.connector.getSessionStatus`.
```
package ee.sk.mydata;

import ee.sk.mid.*;
import ee.sk.mid.rest.dao.MidSessionStatus;
import ee.sk.mid.rest.dao.request.MidAuthenticationRequest;
import ee.sk.mid.rest.dao.response.MidAuthenticationResponse;

public class MidAuthenticator {
    private MidClient midClient;

    public MidAuthenticator() {
        midClient = MidClient.newBuilder()
                .withRelyingPartyUUID("00000000-0000-0000-0000-000000000000")
                .withRelyingPartyName("DEMO")
                .withHostUrl("https://tsp.demo.sk.ee/mid-api")
                .withLongPollingTimeoutSeconds(5)
                .build();
    }

    public MidAuthentication authenticateUser(MidAuthenticationHashToSign authenticationHash,
                                              String identityNumber, String phoneNumber) {
        MidAuthenticationRequest request = MidAuthenticationRequest.newBuilder()
                .withPhoneNumber(phoneNumber)
                .withNationalIdentityNumber(identityNumber)
                .withHashToSign(authenticationHash)
                .withLanguage(MidLanguage.ENG)
                .withDisplayText("Log into self-service?")
                .withDisplayTextFormat(MidDisplayTextFormat.GSM7)
                .build();

        MidAuthenticationResponse response = midClient.getMobileIdConnector().authenticate(request);

        return midClient.createMobileIdAuthentication(getSessionStatus(response), authenticationHash);

    }

    MidSessionStatus getSessionStatus(MidAuthenticationResponse response) {
        return midClient.getSessionStatusPoller()
                .fetchFinalSessionStatus(response.getSessionID(), "/authentication/session/{sessionId}");

    }
}
```
```
public static void main(String[] args) {
        MidAuthenticator authenticator = new MidAuthenticator();
        MidAuthenticationHashToSign authenticationHash = MidAuthenticationHashToSign.generateRandomHashOfDefaultType();
        String identityNumber = null; // Replace value
        String phoneNumber = null; // Replace value
        try {
            authenticator.authenticateUser(authenticationHash, identityNumber, phoneNumber);
        } catch (Exception e) {
            System.err.println("Code executed correctly");
        }
    }
```
**What is the expected behavior?**
`MidSessionStatusPoller.pollForFinalSessionStatus` should not wait longer than 
`longPollingTimeoutSeconds`  value to ask for the sessions status.

**Comment by andrevka** [comment]
It seems to me, as completely normal behavior. I think there's a misunderstanding of the **longPollingTimeoutSeconds** parameter. Currently it is not used on the client as a way to end the request on timeout. Instead it is sent to mid-rest and mid-rest uses it internally to send you a response about the status of your request at most **longPollingTimeoutSeconds** after the request started.

**Comment by aasaru** [comment]
The solution here is to set timeout on the client side as well. Let's implement that.

**Comment by aasaru** [comment]
This was fixed with #18 and will be available in version 1.2
